src/filing/utils/checkFileErrors.js

```diff
@@ -1,31 +1,46 @@
+import detectFileEncoding from 'detect-file-encoding-and-language'
+
 export default function checkFileErrors(file, cb) {
   const fileSlice = file.slice(0, 2)
   const reader = new window.FileReader()
 
   reader.addEventListener('load', () => {
     const firstBytes = reader.result
     const errors = []
-
+    
     if (file && file.size !== undefined && file.name !== undefined) {
-      const notTxt = file.name.split('.').slice(-1)[0].toLowerCase() !== 'txt'
-      if (file.size === 0) {
-        errors.push(
-          'The file you uploaded does not contain any data. Please check your file and re-upload.'
-        )
-      }
-      if (notTxt) {
-        errors.push(
-          'The file you uploaded is not a text file (.txt). Please check your file and re-upload.'
-        )
-      }
-      if(firstBytes !== ('1|') && !notTxt){
-        errors.push('Your file appears to be a text file (.txt) but has invalid content. Please ensure you are uploading a pipe-delimited text file and that your transmittal sheet begins with 1.');
-      }
+      detectFileEncoding(file).then(fileInfo => {
+        if (fileInfo.encoding !== 'UTF-8'){
+          errors.push(
+            'The file you selected is not UTF-8 encoded. Please check your file and re-upload.'
+          )
+          
+          // Skip additional checks to avoid false-positives
+          return cb(errors)
```

Comment:
return here feels awkward, knowing the purpose of it... feels like this section could be refactored a bit; reversing the `if` logic here might flow better; so something like:
```
if (fileInfo.encoding === 'UTF-8') {
    ... // the previous logic
} else {
    cb(...)
}
```
in some of `cb` calls that would only have 1 error, maybe just call `cb(['Your file was...'])` directly, and not have to do `errors.push`.

more of a personal preference addition refactoring, we might be able to do something like:
```
errors = fileSizeCheck(file, errors)
errors = txtCheck(file, errors)
...
```

Reply:
Agreed, that block is hard to decipher.  Refactored.  Thanks for always calling out hard-to-read code!